src/services/mongoose/users.ts

```diff
@@ -85,15 +89,19 @@ export const UsersRepository = (): IUsersRepository => {
   }
 
   const deletePhone = async (id: UserId): Promise<User | RepositoryError> => {
-    const result = await User.findOneAndUpdate(
-      { userId: id },
-      { $unset: { phone: true } },
-      { new: true },
-    )
-    if (!result) {
-      return new RepositoryError("Couldn't delete phone from user")
+    try {
+      const result = await User.findOneAndUpdate(
+        { userId: id },
+        { $unset: { phone: true } },
+        { new: true },
+      )
+      if (!result) {
+        return new CouldNotDeletePhoneFromUser("Couldn't delete phone from user")
```

Comment:
This should be `CouldNotUnsetPhoneFromUser` and don't pass a msg. If you want to pass a message pass it in the overridden constructor where you inherit. But actually the type should be expressive enough to not need a message.